### DISCUSSION TITLE
ModuleNotFoundError: No module named 'ansible.module_utils.common.yaml'

### DISCUSSION THREAD

**Discussion by zep-jonas** [codeaffen/phpipam-ansible-modules#74] [Q&A] [answered]
Hi :)
We use the great phpIPAM for managing our IP documencation.
Now I'm playing with ansible and I'm trying to get this module running.
I installed the module via `ansible-galaxy collection install codeaffen.phpipam` and `pip install phpypam colour geopy inflection ipaddress `.
When trying to run a task with phpipam I receive the following error message:

> ubuntu@ans01:/etc/ansible/playbooks$ ansible-playbook phpipam_reserve_ip_address.yml  -vvv
> ansible-playbook [core 2.12.2]
>   config file = /etc/ansible/ansible.cfg
>   configured module search path = ['/home/ubuntu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
>   ansible python module location = /usr/lib/python3/dist-packages/ansible
>   ansible collection location = /home/ubuntu/.ansible/collections:/usr/share/ansible/collections
>   executable location = /usr/bin/ansible-playbook
>   python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
>   jinja version = 2.10.1
>   libyaml = True
> Using /etc/ansible/ansible.cfg as config file
> host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
> script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
> auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
> Parsed /etc/ansible/hosts inventory source with ini plugin
> Skipping callback 'default', as we already have a stdout callback.
> Skipping callback 'minimal', as we already have a stdout callback.
> Skipping callback 'oneline', as we already have a stdout callback.
> 
> PLAYBOOK: phpipam_reserve_ip_address.yml ******************************************************************************************************************************************************************1 plays in phpipam_reserve_ip_address.yml
> 
> PLAY [Reserve an IP address] ******************************************************************************************************************************************************************************
> TASK [Gathering Facts] ************************************************************************************************************************************************************************************task path: /etc/ansible/playbooks/phpipam_reserve_ip_address.yml:2
> <127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
> <127.0.0.1> EXEC /bin/sh -c 'echo ~ubuntu && sleep 0'
> <127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ubuntu/.ansible/tmp `"&& mkdir "` echo /home/ubuntu/.ansible/tmp/ansible-tmp-1646668235.2833872-1458113-96913956694410 `" && echo ansible-tmp-1646668235.2833872-1458113-96913956694410="` echo /home/ubuntu/.ansible/tmp/ansible-tmp-1646668235.2833872-1458113-96913956694410 `" ) && sleep 0'
> Using module file /usr/lib/python3/dist-packages/ansible/modules/setup.py
> <127.0.0.1> PUT /home/ubuntu/.ansible/tmp/ansible-local-14581084xm4rw3n/tmpa6qwg038 TO /home/ubuntu/.ansible/tmp/ansible-tmp-1646668235.2833872-1458113-96913956694410/AnsiballZ_setup.py
> <127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/ubuntu/.ansible/tmp/ansible-tmp-1646668235.2833872-1458113-96913956694410/ /home/ubuntu/.ansible/tmp/ansible-tmp-1646668235.2833872-1458113-96913956694410/AnsiballZ_setup.py && sleep 0'
> <127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /home/ubuntu/.ansible/tmp/ansible-tmp-1646668235.2833872-1458113-96913956694410/AnsiballZ_setup.py && sleep 0'
> <127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/ubuntu/.ansible/tmp/ansible-tmp-1646668235.2833872-1458113-96913956694410/ > /dev/null 2>&1 && sleep 0'
> ok: [localhost]
> META: ran handlers
> 
> TASK [Reserve IPAM Address] *******************************************************************************************************************************************************************************task path: /etc/ansible/playbooks/phpipam_reserve_ip_address.yml:6
> <127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
> <127.0.0.1> EXEC /bin/sh -c 'echo ~ubuntu && sleep 0'
> <127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ubuntu/.ansible/tmp `"&& mkdir "` echo /home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675 `" && echo ansible-tmp-1646668236.4645762-1458213-272829672950675="` echo /home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675 `" ) && sleep 0'
> Using module file /home/ubuntu/.ansible/collections/ansible_collections/codeaffen/phpipam/plugins/modules/address.py
> <127.0.0.1> PUT /home/ubuntu/.ansible/tmp/ansible-local-14581084xm4rw3n/tmpdx0lfc1p TO /home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py
> <127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/ /home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py && sleep 0'
> <127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py && sleep 0'
> <127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/ > /dev/null 2>&1 && sleep 0'
> The full traceback is:
> Traceback (most recent call last):
>   File "/home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py", line 107, in <module>
>     _ansiballz_main()
>   File "/home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py", line 99, in _ansiballz_main
>     invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
>   File "/home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py", line 47, in invoke_module
>     runpy.run_module(mod_name='ansible_collections.codeaffen.phpipam.plugins.modules.address', init_globals=dict(_module_fqn='ansible_collections.codeaffen.phpipam.plugins.modules.address', _modlib_path=modlib_path),
>   File "/usr/lib/python3.8/runpy.py", line 207, in run_module
>     return _run_module_code(code, init_globals, run_name, mod_spec)
>   File "/usr/lib/python3.8/runpy.py", line 97, in _run_module_code
>     _run_code(code, mod_globals, init_globals,
>   File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
>     exec(code, run_globals)
>   File "/tmp/ansible_codeaffen.phpipam.address_payload_1a3zihxz/ansible_codeaffen.phpipam.address_payload.zip/ansible_collections/codeaffen/phpipam/plugins/modules/address.py", line 129, in <module>     
>   File "<frozen zipimport>", line 259, in load_module
>   File "/tmp/ansible_codeaffen.phpipam.address_payload_1a3zihxz/ansible_codeaffen.phpipam.address_payload.zip/ansible_collections/codeaffen/phpipam/plugins/module_utils/phpipam_helper.py", line 28, in <module>
>   File "/usr/lib/python3/dist-packages/ansible/utils/display.py", line 37, in <module>
>     from ansible import constants as C
>   File "/usr/lib/python3/dist-packages/ansible/constants.py", line 14, in <module>
>     from ansible.config.manager import ConfigManager, ensure_type
>   File "/usr/lib/python3/dist-packages/ansible/config/manager.py", line 21, in <module>
>     from ansible.module_utils.common.yaml import yaml_load
> ModuleNotFoundError: No module named 'ansible.module_utils.common.yaml'
> fatal: [localhost]: FAILED! => {
>     "changed": false,
>     "module_stderr": "Traceback (most recent call last):\n  File \"/home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/ubuntu/.ansible/tmp/ansible-tmp-1646668236.4645762-1458213-272829672950675/AnsiballZ_address.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.codeaffen.phpipam.plugins.modules.address', init_globals=dict(_module_fqn='ansible_collections.codeaffen.phpipam.plugins.modules.address', _modlib_path=modlib_path),\n  File \"/usr/lib/python3.8/runpy.py\", line 207, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.8/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_codeaffen.phpipam.address_payload_1a3zihxz/ansible_codeaffen.phpipam.address_payload.zip/ansible_collections/codeaffen/phpipam/plugins/modules/address.py\", line 129, in <module>\n  File \"<frozen zipimport>\", line 259, in load_module\n  File \"/tmp/ansible_codeaffen.phpipam.address_payload_1a3zihxz/ansible_codeaffen.phpipam.address_payload.zip/ansible_collections/codeaffen/phpipam/plugins/module_utils/phpipam_helper.py\", line 28, in <module>\n  File \"/usr/lib/python3/dist-packages/ansible/utils/display.py\", line 37, in <module>\n    from ansible import constants as C\n  File \"/usr/lib/python3/dist-packages/ansible/constants.py\", line 14, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type\n  File \"/usr/lib/python3/dist-packages/ansible/config/manager.py\", line 21, in <module>\n    from ansible.module_utils.common.yaml import yaml_load\nModuleNotFoundError: No module named 'ansible.module_utils.common.yaml'\n",
>     "module_stdout": "",
>     "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
>     "rc": 1
> }

Ansible Version:
ansible --version
ansible [core 2.12.2]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/ubuntu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/ubuntu/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 2.10.1
  libyaml = True

Phpipam playbook
---
- name: "Reserve an IP address"
  hosts: localhost
  connection: local
  tasks: 
    - name: "Reserve IPAM Address"
      codeaffen.phpipam.address:
        username: "phpipam-api-user"
        password: "xxx"
        server_url: "https://ipam-url"
        address: "10.10.0.1"
        section: "Section from phpIPAM"
        description: "Test-Ansible-Created"
        subnet: "10.10.0.0/24"
        is_gateway: no
        state: present

I can not find a solution for this, can you help me? 
Thank you!

**Comment by cmeissner** [comment, marked as answer]
Hi @zep-jonas 

Thank you for get in touch with us. Unfortunately there is an import inside our module_utils. This is related to early debugging tasks. We used `Display` class from `ansible.utils.display` module. But as modules do not support output stdout this class will not work. Newer versions of Ansible run into the error you describe. 

Currently the CI is running for the next release, where the issue is fixed. As soon as version 1.5.0 is released you should be able to use the collection without any issues from our side.

**Reply by zep-jonas** [reply]
Hallo Christian, 
Vielen Dank für deine Antwort!
